WikiWatershed/mmw-micro
# Initial App Import

## Overview

Imports micro site from Model My Watershed. Sets up basic bundling into `/dist`.
### Notes

In an effort to minimize the difference during import, I've tried to keep the scripts consistent with what they were in Model My Watershed. However, they are now inconsistent with all our other projects which run scripts from within the VM, rather than from outside. I'm making cards to address this in the future.

We probably don't need the `--vendor` flag in the bundling script anymore. That is also a holdover from Model My Watershed and should be removed.

Apparently we minify only JavaScript, not any of the CSS. We should look into minifying CSS. Also, I just copied over the entire SASS directory from Model My Watershed. We can probably lose a lot of that.

Maybe we shouldn't be exporting to the `/dist` folder and serving from there for development? This was the easiest thing to do right now, but I feel `/dist` should only be used for production builds.

Even though this is going to be on a CDN, there's still a _lot_ of files being requested. We should look into using CSS rather than Retina.js for serving higher resolution images to reduce the number of requests, and maybe also look into CSS sprites if that is not enough.
## Testing Instructions

Check out this branch. Reload Vagrant so it picks up on new server port mapping. Then run the bundle script:

``` bash
$ ./scripts/bundle.sh --vendor
```

Open [localhost:8000](http://localhost:8000/) in a browser and test that the app works identically to the one on [staging](http://staging.app.wikiwatershed.org/micro), and that there are no errors in the console.

Connects https://github.com/WikiWatershed/model-my-watershed/issues/1483
